ModelCatalogueGenomicsPlugin/grails-app/controllers/org/modelcatalogue/core/genomics/GenomicsController.groovy

```diff
@@ -1,12 +1,71 @@
 package org.modelcatalogue.core.genomics
 
+import groovy.util.logging.Log4j
+import org.modelcatalogue.core.AbstractCatalogueElementController
+import org.modelcatalogue.core.Asset
 import org.modelcatalogue.core.DataClass
+import org.modelcatalogue.core.api.ElementStatus
 import org.springframework.http.HttpStatus
 
 /**
  * Controller for GEL specific reports.
  */
-class GenomicsController {
+@Log4j
+class GenomicsController extends AbstractCatalogueElementController<DataClass> {
+
+    def gelJsonService
+    def executorService
+    def auditService
+    def springSecurity2SecurityService
+
+
+    GenomicsController() {
+        super(DataClass, false)
+    }
+
+    def exportRareDiseaseHPOAndClinicalTests() {
+        DataClass model = DataClass.get(params.id)
+
+
+        def asset = new Asset(
```

Comment:
please, use `assetService.storeReportAsAsset` method which does all the `Asset` handling and auditing for you.